packages/app/src/systems/Asset/services/assets.ts

```diff
@@ -64,6 +64,12 @@ export class AssetService {
 
   static async addAssets(input: AssetInputs['addAssets']) {
     return db.transaction('rw', db.assets, async () => {
+      const someNameUndefined = input.data.some((asset) => {
+        asset.name === undefined;
+      });
+      if (someNameUndefined) {
+        throw new Error('Asset.name is undefined');
+      }
```

Comment:
This should be implemented inside the method `validateAddAssets` like the other validations: https://github.com/FuelLabs/fuels-wallet/blob/21540c5ddc921f052f5c5fde040add5b7cf3cd7c/packages/app/src/systems/CRX/background/services/BackgroundService.ts#L328